python/test/segy.py

```diff
@@ -937,6 +937,108 @@ def test_write_header_update_atomic(small):
         assert header == f.header[10]
 
 
+def test_field_types_read():
+    with segyio.open(testdata / 'decrement.sgy', "r", ignore_geometry=True) as f:
+        bdec = f.bin
+        tdec = f.header[3]
+
+        assert bdec[BinField.NrTracesInStream] == 18446744073709551572
+        assert bdec[BinField.JobID] == 2147483647
+        assert bdec[BinField.SamplesOriginal] == 65526
+        assert bdec[BinField.VerticalSum] == 32755
+        assert bdec[BinField.SEGYRevisionMinor] == 217
+        assert bdec[BinField.ExtIntervalOriginal] == 1125899906842593.0
+
+        assert tdec[TraceField.INLINE_3D] == 2147483355
+        assert tdec[TraceField.TRACE_SAMPLE_COUNT] == 65382
+        assert tdec[TraceField.DelayRecordingTime] == 32627
+
+    with segyio.open(testdata / 'increment.sgy', "r", ignore_geometry=True) as f:
+        binc = f.bin
+        tinc = f.header[3]
+
+        assert binc[BinField.NrTracesInStream] == 43
+        assert binc[BinField.JobID] == -2147483648
+        assert binc[BinField.SamplesOriginal] == 8
+        assert binc[BinField.VerticalSum] == -32756
+        assert binc[BinField.SEGYRevisionMinor] == 37
+        assert binc[BinField.ExtIntervalOriginal] == -1125899906842593.0
+
+        assert tinc[TraceField.INLINE_3D] == -2147483356
+        assert tinc[TraceField.TRACE_SAMPLE_COUNT] == 152
+        assert tinc[TraceField.DelayRecordingTime] == -32628
+
+
+@tmpfiles(testdata / 'small.sgy')
+def test_field_types_write(tmpdir):
+    updpath = tmpdir / 'small.sgy'
+
+    with segyio.open(updpath, ignore_geometry=True) as f:
+        orig_binh = f.bin
+        orig_trh = f.header[0]
+
+    with segyio.open(updpath, mode='r+', ignore_geometry=True) as f:
+        bupd = {
+            BinField.NrTracesInStream: 18446744073709551615,
+            BinField.JobID: 2147483647,
+            BinField.SamplesOriginal: 65535,
+            BinField.VerticalSum: -32768,
+            BinField.SEGYRevisionMinor: 255,
+            BinField.ExtIntervalOriginal: 10.125,
+        }
+
+        tupd = {
+            TraceField.INLINE_3D: -2147483648,
+            TraceField.TRACE_SAMPLE_COUNT: 65535,
+            TraceField.DelayRecordingTime: 32767,
+        }
```

Comment:
I would consider moving the definition of `tupd` and `bupd` of the `with ... segyio.open ... as f` scope (just in front to it). `tupd` and `bupd` are used in this and the next `with ... segyio.open ... as f` scope. I was first confused where `tupd` and `bupd` came from in the second scope that compares the headers and checks that the values were written. 

Misunderstanding this might be my mistake though. I am too used to C++ in which variables within the scope would be invalidated outside of it.